source/PreferencesPanel.cpp

```diff
@@ -1306,13 +1528,63 @@ void PreferencesPanel::HandleConfirm()
 	case 'p':
 		Plugins::TogglePlugin(selectedPlugin);
 		break;
+	case 'i':
+		if(latestPlugin && latestPlugin->url.size() && !latestPlugin->installed)
+			installFeedbacks.emplace_back(Plugins::Install(latestPlugin));
+		else if(latestPlugin && latestPlugin->url.size() && latestPlugin->outdated)
+			installFeedbacks.emplace_back(Plugins::Update(latestPlugin));
+		break;
 	default:
 		break;
 	}
 }
 
 
 
+void PreferencesPanel::ProcessPluginIndex()
+{
+	installFeedbacks.emplace_back(async(launch::async, [&]() noexcept -> void {
+		GetUI()->Push(new Dialog("Downloading plugin index, please wait."));
+		if(!Plugins::Download(PLUGIN_LIST_URL, Files::Config() + "plugins.json"))
+		{
+			GetUI()->Pop(GetUI()->Top().get());
+			GetUI()->Push(new Dialog(this, &PreferencesPanel::ProcessPluginIndex,
+				"Failed to download plugin index, try again?"));
+			return;
+		}
+		ifstream pluginlistFile(Files::Config() + "plugins.json");
+		nlohmann::json pluginInstallList = nlohmann::json::parse(pluginlistFile);
+		pluginInstallPages = ((pluginInstallList.size() - (pluginInstallList.size() % MAX_PLUGIN_INSTALLS_PER_PAGE))
+			/ MAX_PLUGIN_INSTALLS_PER_PAGE) + (pluginInstallList.size() % MAX_PLUGIN_INSTALLS_PER_PAGE > 0);
```

Comment:
Maybe just do a floating point division and round it up?

Reply:
```suggestion
		pluginInstallPages = ceil(static_cast<float>(pluginInstallList.size()) / static_cast<float>(MAX_PLUGIN_INSTALLS_PER_PAGE));
```